05.wc/wc.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, filenames = parse_options
+  file_stats = collect_file_stats(filenames)
+  total_stats = calculate_total_stats(file_stats)
+  max_widths = calculate_max_widths(total_stats)
+  print_file_stats(file_stats, max_widths, options)
+  print_file_stats([total_stats], max_widths, options) if filenames.size > 1
+end
+
+def parse_options
+  options = {}
+  OptionParser.new do |opts|
+    opts.on('-l') { options[:lines] = true }
+    opts.on('-w') { options[:words] = true }
+    opts.on('-c') { options[:bytes] = true }
+  end.parse!
+  options = { bytes: true, lines: true, words: true } if options.empty?
+  filenames = ARGV.empty? ? [''] : ARGV
+  [options, filenames]
+end
+
+def collect_file_stats(filenames)
+  filenames.map do |filename|
+    input = filename.empty? ? ARGF.read : File.read(filename)
```

Comment:
```suggestion
    text = filename.empty? ? ARGF.read : File.read(filename)
```

文字列ですよ、ということがわかる名前の方がベターかな〜